src/plugins/data/public/query/query_string/query_string_manager.ts

```diff
@@ -81,7 +86,13 @@ export class QueryStringManager {
     ) {
       const language = this.languageService.getLanguage(defaultLanguageId);
       const newQuery = { ...query, dataset: defaultDataset };
-      const newQueryString = language?.getQueryString(newQuery) || '';
+
+      const defaultDatasetQueryString = this.getDefaultDatasetQueryString(
+        defaultDataset,
+        newQuery.language
+      );
+      const newQueryString =
+        defaultDatasetQueryString ?? (language?.getQueryString(newQuery) || '');
```

Comment:
```suggestion
        this.getDefaultDatasetQueryString(
        defaultDataset,
        newQuery.language
      ) || language?.getQueryString(newQuery) || '';
```

for consistency

Reply:
perhaps we need a default query string at the workspace level but yeah for DQL and Lucene it's a valid query and the original behavior.

Reply:
No op